### PR TITLE
Tree routines

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
Here is a summary of the status of all the routines.

<center>

| Routine  | dummy | qw5q _gold - qblox | qw25q - qm | tii1q_b1 - rfsoc  | iqm5q - zurich
|---|:---:|:---:|:---:|:---:|:---:|
| resonator spectroscopy  | ✔️  | ✔️ | ✔️  |✔️    |   |
| resonator punchout  | ✔️   | ✔️   | ✔️  | ✔️  |   |
| resonator punchout att  |    | ✔️   |   |   |   |
| resonator flux dependence  |✔️   | ✔️  |   |  |  | 
| qubit spectroscopy  | ✔️   |  ✔️ |   | ✔️  |   |
| qubit flux dependence  | ✔️   | ✔️  |   |   |   |
|  rabi amplitude | ✔️   | ✔️  |   | ✔️  |   |
|  rabi length | ✔️   |  ✔️  |   |✔️   |   |
|  t1 | ✔️   |  ✔️ |   | ✔️  |    |
|  ramsey |✔️   | ✔️   |   | ✔️  |   |
|  ramsey detuned |✔️    | ✔️  |   | ✔️  |   |   
|  single shot calibration | ✔️   | ✔️  |   | ✔️  |   |   
|  allXY | ✔️   | ✔️  |   | ✔️  |   |
|  drag pulse |✔️    | ✔️  |   | ✔️  |   |   
|  spin echo | ✔️   |   |   |✔️   |   |
|  rb |  |   |   |   |   |
|  two qubit gates routines | |   |   |   |   |  

</center>

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibocal/protocols/characterization/resonator_punchout_attenuation.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    )
+    freq_sweeper = Sweeper(
+        Parameter.frequency,
+        delta_frequency_range,
+        [ro_pulses[qubit] for qubit in qubits],
+    )
+
+    # attenuation
+    attenuation_range = np.arange(params.min_att, params.max_att, params.step_att)
+    amp_sweeper = Sweeper(
```
Comment (DavidSarlle) [inline src/qibocal/protocols/characterization/resonator_punchout_attenuation.py:79]: @Edoardo-Pedicillo can you change the name of amp_sweeper to att_sweeper?

File: src/qibocal/protocols/characterization/resonator_punchout_attenuation.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        Parameter.attenuation, attenuation_range, qubits=[qubit for qubit in qubits]
+    )
+
+    # create a DataUnits object to store the results,
+    # DataUnits stores by default MSR, phase, i, q
+    # additionally include resonator frequency and attenuation
+    data = ResonatorPunchoutAttenuationData()
+
+    # repeat the experiment as many times as defined by software_averages
+    amps = np.repeat(attenuation_range, len(delta_frequency_range))
```
Comment (DavidSarlle) [inline src/qibocal/protocols/characterization/resonator_punchout_attenuation.py:89]: also here amps by atts

**Comment by Edoardo-Pedicillo** [comment]
Done. Thank @DavidSarlle!

**Comment by andrea-pasquale** [comment]
Thanks @DavidSarlle for https://github.com/qiboteam/qibocal/pull/290/commits/aa16ecd46a9f0e7bd33a1c34f7a4af8bd8795ad4 and https://github.com/qiboteam/qibocal/pull/290/commits/dcac855b35e5b8f4e73003ff2713b9cb3eea8eb9.
I had the fix locally and I was about to push it.

**Comment by DavidSarlle** [comment]
> Thanks @DavidSarlle for [aa16ecd](https://github.com/qiboteam/qibocal/commit/aa16ecd46a9f0e7bd33a1c34f7a4af8bd8795ad4) and [dcac855](https://github.com/qiboteam/qibocal/commit/dcac855b35e5b8f4e73003ff2713b9cb3eea8eb9). I had the fix locally and I was about to push it.

Ok. Sorry. They were quick fixes and I asked to @Edoardo-Pedicillo to pcommit the fixes ddirectly. I have also tested with qblox  and qw5q.

**Comment by andrea-pasquale** [comment]
> > Thanks @DavidSarlle for [aa16ecd](https://github.com/qiboteam/qibocal/commit/aa16ecd46a9f0e7bd33a1c34f7a4af8bd8795ad4) and [dcac855](https://github.com/qiboteam/qibocal/commit/dcac855b35e5b8f4e73003ff2713b9cb3eea8eb9). I had the fix locally and I was about to push it.
> 
> Ok. Sorry. They were quick fixes and I asked to @Edoardo-Pedicillo to pcommit the fixes ddirectly. I have also tested with qblox and qw5q.

Perfect. No problem. Thank you!

**Comment by andrea-pasquale** [comment]
@Edoardo-Pedicillo @DavidSarlle if we are happy with the current status of this PR we could merge it to #279.
Let me know what you think.

**Comment by DavidSarlle** [comment]
@andrea-pasquale from the point of view of the code it is ok from me. On the other hand, I am still debugging the results obtained from some routines with qblox to be sure that they are correct after all the changes introduced in the driver and qibocal. But I can do it from #279 . 

You can see the status of tests here: https://github.com/qiboteam/qibolab/pull/370.

**Review by DavidSarlle — APPROVED** [review]

**Comment by Edoardo-Pedicillo** [comment]
we can merge it

**Comment by andrea-pasquale** [comment]
Thanks @DavidSarlle and @Edoardo-Pedicillo.

**Pull request merged**